### PR TITLE
file batch source enhancements

### PR DESCRIPTION
**Pull request by albertshau**
Adding macro support to the FileBatchSource, and adding a field
to its output that contains the split path the record was read
from.

### CODE REVIEW

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FileBatchSource.java [hunk 1]
```diff
@@ -228,7 +222,7 @@ public FileBatchConfig() {
       super("");
       this.fileSystemProperties = GSON.toJson(ImmutableMap.<String, String>of());
       this.fileRegex = ".*";
-      this.inputFormatClass = CombineTextInputFormat.class.getName();
+      this.inputFormatClass = TextInputFormat.class.getName();
```
Comment (gokulavasan) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FileBatchSource.java:225]: Any reason behind this change? I guess it should be have TextInputFormat to begin with.

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/StructuredRecordFileInputFormat.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Wrapper around a FileInputFormat that attempts to capture the path each record was read from.
+ */
+public class StructuredRecordFileInputFormat extends FileInputFormat<NullWritable, StructuredRecord> {
+  public static final Schema SCHEMA = Schema.recordOf(
+    "fileRecord",
+    Schema.Field.of("offset", Schema.nullableOf(Schema.of(Schema.Type.LONG))),
```
Comment (gokulavasan) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/StructuredRecordFileInputFormat.java:43]: Do we document anywhere that this is the schema of the records emitted by this source?

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/StructuredRecordFileInputFormat.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+      return NullWritable.get();
+    }
+
+    @Override
+    public StructuredRecord getCurrentValue() throws IOException, InterruptedException {
+      Object key = delegate.getCurrentKey();
+      Object val = delegate.getCurrentValue();
+      return StructuredRecord.builder(SCHEMA)
+        .set("path", path)
+        .set("offset", key instanceof LongWritable ? ((LongWritable) key).get() : null)
```
Comment (gokulavasan) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/StructuredRecordFileInputFormat.java:131]: Is the offset, the offset within that particular file or is it offset across all files put together?

**Comment by gokulavasan** [comment]
Few comments. Can we make sure this doesn't break plugins that extend FileBatchSource, ex FTPSource etc?

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/StructuredRecordFileInputFormat.java (hunk 3 above)
Comment (nitinmotgi) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/StructuredRecordFileInputFormat.java:131]: It's the offset in the file.

**Comment by albertshau** [comment]
macros are being added as a generic capability for all of hydrator instead of using this approach. closing.

**Pull request closed without merging**